src/codec/data_storage/data_block.rs

```diff
@@ -0,0 +1,276 @@
+use elliptic_curve::rand_core::CryptoRngCore;
+use futures::{AsyncWrite, AsyncWriteExt};
+use winnow::binary::le_u8;
+use winnow::token::literal;
+use winnow::Parser;
+
+use super::data_options::DataOptions;
+use super::encrypted_data_chunk::EncryptedDataChunk;
+use crate::codec::crypto::{AuthenticationTag, Nonce};
+use crate::codec::header::BANYAN_DATA_MAGIC;
+use crate::codec::meta::{BlockSize, BlockSizeError};
+use crate::codec::{Cid, ParserResult, Stream};
+use crate::utils::std_io_err;
+
+use std::sync::{Arc, RwLock};
+
+pub struct DataBlock {
+    data_options: DataOptions,
+    cid: Arc<RwLock<Option<Cid>>>,
+    contents: Vec<EncryptedDataChunk>,
+}
+
+impl DataBlock {
+    /// The amount of payload a small encrypted block can hold.
+    #[allow(clippy::identity_op)]
+    pub const SMALL_ENCRYPTED_SIZE: usize =
+        262_144 - (1 * Nonce::size() + AuthenticationTag::size());
+
+    /// The amount of payload a standard encrypted block can hold.
+    pub const STANDARD_ENCRYPTED_SIZE: usize =
+        134_217_728 - (64 * Nonce::size() + AuthenticationTag::size());
+
+    pub fn base_chunk_size(&self) -> usize {
+        self.data_options().chunk_size() as usize
+    }
+
+    pub fn chunk_size(&self) -> usize {
+        let mut base_chunk_size = self.base_chunk_size();
+
+        // length bytes
+        base_chunk_size -= 8;
```

Comment:
Good yeah, so the size is being taking into account here but not in those constants (I'd be happy if the constants went away and we were calculating the other hardcoded values directly)